nuc_morph_analysis/analyses/linear_regression/linear_regression_workflow.py

```diff
@@ -0,0 +1,259 @@
+                "r2",
+                "explained_variance",
+                "neg_mean_absolute_error",
+                "max_error",
+                "neg_mean_squared_error",
+                "neg_mean_absolute_percentage_error",
+            ],
+            return_train_score=True,
+        )
+
+        # Save test r^2 and test MSE to dataframe
+        range_test_scores = [round(i, 2) for i in cv_model["test_r2"]]
+        range_errors = [
+            round(i, 2) for i in cv_model["test_neg_mean_squared_error"]
+        ]
+        test_sc = pd.DataFrame()
+        test_sc[r"Test r$^2$"] = range_test_scores
+        test_sc["Test MSE"] = range_errors
+        test_sc["alpha"] = this_alpha
+        all_test_sc.append(test_sc)
+
+        # Save coeffs to dataframe
+        coefs = pd.DataFrame(
+            [model[1].coef_ for model in cv_model["estimator"]], columns=cols
+        )
+
+        all_coefs.append(coefs)
+        all_coefs = pd.concat(all_coefs, axis=0).reset_index(drop=True)
+        all_coefs["alpha"] = this_alpha
+        all_coef_alpha.append(all_coefs)
+
+    # Get test scores for all alpha
+    all_test_sc = pd.concat(all_test_sc, axis=0).reset_index(drop=True)
+    all_test_sc["Test MSE"] = -all_test_sc["Test MSE"]
+    all_test_sc.to_csv(save_path / "mse.csv")
+
+    # Get coeffs for all alpha
+    all_coef_alpha = pd.concat(all_coef_alpha, axis=0).reset_index(drop=True)
+    all_coef_alpha = all_coef_alpha.melt(
+        id_vars=["alpha"],
+        var_name="Column",
+        value_name="Coefficient Importance",
+    ).reset_index(drop=True)
+    all_coef_alpha.to_csv(save_path / "coefficients.csv")
+
+    # Get permutation scores and p values for all alpha
+    all_perms = pd.DataFrame(all_perms).reset_index(drop=True)
+    all_perms.to_csv(save_path / "perm_scores.csv")
+
+    # Save coefficient plot for max alpha value
+    save_plots(all_coef_alpha, all_test_sc, all_perms, save_path)
+
+    return all_coef_alpha, all_test_sc, all_perms
+
+def save_plots(all_coef_alpha, all_test_sc, all_perms, save_path):
+
+    # subset to max alpha
+    max_alpha = all_coef_alpha['alpha'].max()
+    all_coef_alpha = all_coef_alpha.loc[all_coef_alpha['alpha'] == max_alpha].reset_index(drop=True)
+    all_test_sc = all_test_sc.loc[all_test_sc['alpha'] == max_alpha].reset_index(drop=True)
+    all_perms = all_perms.loc[all_perms['alpha'] == max_alpha].reset_index(drop=True)
+
+    p_value = round(all_perms['p_value'].item(), 3)
+    test_r2_mean = round(all_test_sc['Test r$^2$'].mean(), 2)
+    test_r2_std = round(all_test_sc['Test r$^2$'].std()/2, 2)
+
+
+    g = sns.catplot(
+        data=all_coef_alpha,
+        x='Column',
+        y="Coefficient Importance",
+        kind="bar",
+        errorbar="sd",
+        aspect=1.5,
+        height=4,
+    )
+    g.fig.subplots_adjust(top=0.8) # adjust the Figure in rp
+    g.fig.suptitle(f'p-value {p_value}, test r^2 {test_r2_mean}+-{test_r2_std}')
+    g.set_xticklabels(rotation=90)
```

Comment:
Take or leave at the current moment. Once we are at the stage of making the figures pretty we can use insert the name of the column used in the paper like this:

```
from nuc_morph_analysis.lib.visualization.plotting_tools import get_plot_labels_for_metric

label_list = [get_plot_labels_for_metric(col)[1] for col in all_coef_alpha['Column'].unique()]
g.set_xticklabels(label_list, rotation=90)
```